From 1fac0422f40e65229d9a9aa2ce2643cce797c0e5 Mon Sep 17 00:00:00 2001
From: SongFerngWang <songferngwang@google.com>
Date: Wed, 5 May 2021 21:33:00 +0800
Subject: [PATCH] SubscriptionGroup is exposed to unprivileged callers

SubscriptionInfo.mGroupUUID is not cleared in
conditionallyRemoveIdentifiers if the caller only has READ_PHONE_STATE
(based on a check to checkReadPhoneState) and not READ_DEVICE_IDENTIFIERS.
Bug: 181053462
Test: atest SubscriptionManagerTest

Change-Id: Ic2b62523330dc6e2169ad851715c4ab3da3b29cf
Merged-In: Ic2b62523330dc6e2169ad851715c4ab3da3b29cf
(cherry picked from commit 121c969e22950cd966bb99b101c1e7e50084ae3c)
---
 telephony/java/android/telephony/SubscriptionInfo.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/telephony/java/android/telephony/SubscriptionInfo.java b/telephony/java/android/telephony/SubscriptionInfo.java
index 0ee6568b6430..90d7a161767c 100644
--- a/telephony/java/android/telephony/SubscriptionInfo.java
+++ b/telephony/java/android/telephony/SubscriptionInfo.java
@@ -566,6 +566,13 @@ public class SubscriptionInfo implements Parcelable {
         return mGroupUUID;
     }
 
+    /**
+     * @hide
+     */
+    public void clearGroupUuid() {
+        this.mGroupUUID = null;
+    }
+
     /**
      * @hide
      */
-- 
2.17.1

